src/util-exception-policy.c

```diff
@@ -73,6 +75,83 @@ void ExceptionPolicyApply(Packet *p, enum ExceptionPolicy policy, enum PacketDro
     SCLogDebug("end");
 }
 
+/* Should I make a separate enum for this? With a separate function?
+   This seems to make sense in my head right now, as then things would be more logically containted,
+   and then I could call the master-switch function from within the ExceptionPolicyParse function...
+ */
+
+void setMasterExceptionPolicy()
+{
+    const char *switch_str = NULL;
+    if (EngineModeIsIPS()) {
+        g_eps_master_switch = EXCEPTION_POLICY_SWITCH_DEFAULT;
+        if (ConfGet("exception-policy-master-switch", &switch_str) == 1 && switch_str != NULL) {
+            if (strcmp(switch_str, "auto") == 0) {
+                SCLogConfig("Exception Policies set to 'drop-packet/drop-flow' via "
+                            "master switch: 'auto' mode");
+                g_eps_master_switch = EXCEPTION_POLICY_SWITCH_DEFAULT;
+            } else if (strcmp(switch_str, "performance") == 0) {
+                SCLogConfig("Exception Policies set to 'bypass-flow' via master switch: "
+                            "'performance' mode");
+                g_eps_master_switch = EXCEPTION_POLICY_SWITCH_PERFORMANCE;
+            } else if (strcmp(switch_str, "disabled") == 0) {
+                SCLogConfig("master switch for exception policies is disabled, "
+                            "exception policies should be configured individually.");
+                g_eps_master_switch = EXCEPTION_POLICY_SWITCH_DISABLED;
+            } else {
+                FatalErrorOnInit(SC_ERR_INVALID_ARGUMENT,
+                        "\"%s\" is not a valid master switch value for the exception policy."
+                        " Valid options are auto, performance or disabled.",
+                        switch_str);
+            }
+        } else {
+            /* not enabled, we won't change the master exception policy,
+             for now */
+            SCLogWarning(SC_ERR_CONF_YAML_ERROR,
+                    "exception-policy-master-switch value not set, so ignoring it."
+                    " This behavior will change in Suricata 8, so please update your"
+                    " config. See ticket #5219 for more details.");
+            g_eps_master_switch = EXCEPTION_POLICY_SWITCH_DISABLED;
+        }
+    }
+}
+
+/** brief Set a master Exception Policy, if one has been defined
+ *
+ *  \retval true if the master Exception Policy has been set
+ *  \retval false if the master switch is disabled and no policy was set
+ */
+static bool getMasterExceptionPolicy(const char *option, enum ExceptionPolicy *policy)
```

Comment:
might also extend the ExceptionPolicy to include a EXCEPTION_POLICY_NOT_SET and just return that

Reply:
Can you think of any differences between EXCEPTION_POLICY_IGNORE and what EXCEPTION_POLICY_NOT_SET would do? I'm thinking that it's the same, since we use 'ignore' when the user doesn't want to apply exception policies...